rero_ils/modules/stats/api.py

```diff
@@ -0,0 +1,277 @@
+
+        point in time
+        :param library_pid: string - the library to filter with
+        :return: the number of matched items
+        :rtype: integer
+        """
+        # can be done using the facet
+        return ItemsSearch().filter(
+            'term', library__pid=library_pid).count()
+
+    def number_of_deleted_items(self, library_pid):
+        """Number of items to my library during the specified timeframe.
+
+        :param library_pid: string - the library to filter with
+        :return: the number of matched deleted items
+        :rtype: integer
+        """
+        return RecordsSearch(index=LoanOperationLog.index_name)\
+            .filter('range', date=self.date_range)\
+            .filter('term', operation='delete')\
+            .filter('term', record__type='item')\
+            .filter('term', library__pid=library_pid)\
+            .count()
+
+    def number_of_new_items(self, library_pid):
+        """Number of new created items during the specified timeframe.
+
+        :param library_pid: string - the library to filter with
+        :return: the number of matched newly created items
+        :rtype: integer
+        """
+        # can be done using the facet or operation logs
+        return ItemsSearch()\
+            .filter('range', _created=self.date_range)\
+            .filter('term', library__pid=library_pid).count()
+
+    def number_of_checkins(self, library_pid):
+        """Number of checkins of a library during the specified timeframe.
+
+        :param library_pid: string - the library to filter with
+        :return: the number of matched checkins
+        :rtype: integer
+        """
+        return RecordsSearch(index=LoanOperationLog.index_name)\
+            .filter('range', date=self.date_range)\
+            .filter('term', loan__trigger='checkin')\
+            .filter('term', loan__item__library_pid=library_pid)\
+            .count()
+
+    def number_of_requests(self, library_pid):
+        """Number of renewals of a library during the specified timeframe.
+
+        :param library_pid: string - the library to filter with
+        :return: the number of matched requests
+        :rtype: integer
+        """
+        return RecordsSearch(index=LoanOperationLog.index_name)\
+            .filter('range', date=self.date_range)\
+            .filter('term', loan__trigger='request')\
+            .filter('term', loan__item__library_pid=library_pid)\
+            .count()
+
+    def number_of_new_patrons(self, organisation_pid):
+        """New patrons for an organisation during the specified timeframe.
+
+        :param organisation_pid: string - the organisation to filter with
+        :return: the number of matched newly created patrons
+        :rtype: integer
+        """
+        return PatronsSearch()\
+            .filter('range', _created=self.date_range)\
+            .filter('term', organisation__pid=organisation_pid)\
+            .count()
```

Comment:
What's happens, if we create a Patron at Day0 and delete it at Day+1. It doesn't take account for stats. Is it the correct way ?

Reply:
I do not know. This is not used for billing. Miguel did a review of this. The other approach is to use operation logs which is not enable for patrons for now.

Reply:
Regarding **billing**, we are aware that resources such as patrons, librarians, libraries and documents that are created then deleted within a billing period (a quarter) are not taken into account. This should be ok, at least for now.
But for for the purpose of **operation statistics in general**, it might be useful to take into account such creations and deletions. Whether that is an actual use case or not, is a matter that should be discussed with the PO.

Reply:
Deletion of a patron is rare, and mostly done automatically by us (the Central office) for patrons who are inactive for a long time (3-5 years). So operation logs for patrons are not a priority for now!

Reply:
Ok thus I leave like this.